sdk/search/azure-search/src/main/java/com/azure/search/util/ContinuablePagedIterable.java

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search.util;
+
+import com.azure.core.util.IterableStream;
+import com.azure.core.util.paging.ContinuablePage;
+import com.azure.core.util.paging.ContinuablePagedFlux;
+
+import java.util.stream.Stream;
+
+/**
+ * This class provides utility to iterate over {@link ContinuablePage} using {@link Stream}
+ * {@link Iterable} interfaces.
+ *
+ * @param <C> the type of the continuation token
+ * @param <T> The type of elements in a {@link ContinuablePage}
+ * @param <P> The {@link ContinuablePage} holding items of type {@code T}.
+ *
+ * @see IterableStream
+ * @see ContinuablePagedFlux
+ */
+public abstract class ContinuablePagedIterable<C, T, P extends ContinuablePage<C, T>> extends IterableStream<T> {
```

Comment:
This class will be removed once this is merged into Azure Core, this was added into Search directly to unblock this change.

Reply:
Do we have it for preview 1? Can u put it under impl?

Reply:
The changes for this in Azure Core should be merged before preview 1 ships. This can't be under implementation as it is a super class to a public API and we don't want to have any classes in the hierarchy for a public API be non-public API.